lib/qam.pm

```diff
@@ -81,6 +81,28 @@ sub add_test_repositories {
     my @repos = split(/,/, get_var('MAINT_TEST_REPO', ''));
     my $gpg = get_var('BUILD') =~ m/^MR:/ ? "-G" : "";
     my $system_repos = script_output('zypper lr -u');
+
+    if (is_sle('=12-SP2')) {
+        my $arch = get_var('ARCH');
+        my $url = "http://dist.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP2-LTSS-ERICSSON/$arch/update/";
+        # don't add repo when it's already present
+        unless ($system_repos =~ /$url/) {
+            zypper_call("--no-gpg-checks ar -f $gpg $url '12-SP2-LTSS-ERICSSON-Updates'");
+        }
+    }
+    if (is_sle('=12-SP3')) {
+        my $arch = get_var('ARCH');
+        my $url = "http://dist.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP3-LTSS-TERADATA/$arch/update/";
+        # don't add repo when it's already present
+        unless ($system_repos =~ /$url/) {
+            zypper_call("--no-gpg-checks ar -f $gpg $url '12-SP3-LTSS-TERADATA-Updates'");
+        }
+    }
+    # shim update will fail with old grub2 due to old signature
+    if (check_var('MACHINE', 'uefi')) {
```

Comment:
How about:
if (get_var('MACHINE') =~ /uefi/)

Reply:
We have other machine types like `uefi-2G`

Reply:
I guess I will change to proposal or to look for `UEFI`.